darshan-runtime/lib/darshan-hdf5.c

```diff
@@ -116,17 +116,19 @@ static int my_rank = -1;
 #define HDF5_LOCK() pthread_mutex_lock(&hdf5_runtime_mutex)
 #define HDF5_UNLOCK() pthread_mutex_unlock(&hdf5_runtime_mutex)
 
+#define HDF5_WTIME() \
+    __darshan_disabled ? 0 : darshan_core_wtime();
+
 /*********************************************************
  *        Wrappers for H5F functions of interest         * 
  *********************************************************/
 
 #define H5F_PRE_RECORD() do { \
-    HDF5_LOCK(); \
-    if(!darshan_core_disabled_instrumentation()) { \
+    if(!__darshan_disabled) { \
```

Comment:
Actually, maybe a slightly cleaner variation of the above would be to modify `darshan_core_register_module()` so that it can report failure, and have all of the modules check for success before allocating a module runtime structure.

The safety check to prevent re-initialization after shutdown has started could then be put in `darshan_core_register_module()` so it's not duplicated in N modules.

Reply:
After further discussion: the modules are already protected against re-registration, because the core code sets the `__darshan_core` variable to NULL (preventing further registrations)  before invoking any module output routines.

There was,  however, still a possibility of modules updating counters in between the module `_output()` and `_cleanup()` routines, which could possibly cause inconsistent results in the log.  This is fixed now in 9c86c8d.